operator/controllers/module_catalog_controller.go

```diff
@@ -39,10 +42,9 @@ func (r *ModuleTemplateReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		// on deleted requests.
 		if k8serrors.IsNotFound(err) {
 			logger.Info(req.NamespacedName.String() + " got deleted!")
-			return ctrl.Result{}, nil
+			return ctrl.Result{}, catalogSync.Cleanup(ctx)
```

Comment:
Do we have race condition when both controller handle kyma resource? I'm not sure how exactly it behaviors, but I assume when the kyma out of worker queue, both controller received it. Then imaging a kyma in deleting state enter into reconcile, first in the module_catalog_controller, the `Get` request still can fetch this kyma, then in parallel, in kyma_controller, the finalizer get removed, then this kyma get deleted by apiserver, in this case, it will not enter into worker queue again, right? Then how this catalog cleanup can be triggered?

Reply:
In this case at most it will be deferred by the cache interval of the informer cache, which by default is set to 60 minutes in kubebuilder projects. In this case the delete is triggered and then after an hour the cache will be reset leading to another call that will trigger the not found case on the module catalog. In this case most likely the cluster is gonna be deprovisioned before we can cleanup the catalog though.

Reply:
Let's merge it first, I'll try to do a simulate for this case, it's kind of interesting to me.